filters/f_auto_filters.c

```diff
@@ -45,14 +45,17 @@ static void deint_process(struct mp_filter *f)
         return;
     }
 
+    struct mp_image *img = frame.data;
+    bool interlaced = img->fields & MP_IMGFIELD_INTERLACED ||
+                      img->fields & MP_IMGFIELD_TOP_FIRST ||
+                      img->fields & MP_IMGFIELD_REPEAT_FIRST;
```

Comment:
[`MP_IMGFIELD_TOP_FIRST` is only set if `MP_IMGFIELD_INTERLACED` is also set](https://github.com/FFmpeg/FFmpeg/blob/master/libavutil/frame.h#L638-L639), so we can drop that. And `MP_IMGFIELD_REPEAT_FIRST` will [lead to a ton of false positives](https://github.com/FFmpeg/FFmpeg/blob/master/libavutil/frame.h#L516) which imo we want to avoid.

We only need to check if the frame is interlaced or not, then let ffmpeg figure out how exactly to deinterlace that frame, so I think only checking for `MP_IMGFIELD_INTERLACED` is enough.

Reply:
Reading the actual headers definitely helps.